net/respondd/src/respondd.c

```diff
@@ -361,13 +408,118 @@ static void serve(int sock) {
 	}
 
 	if (output) {
-		if (sendto(sock, output, output_bytes, 0, (struct sockaddr *)&addr, addrlen) < 0)
+		if (sendto(sock, output, output_bytes, 0, addr, sizeof(struct sockaddr_in6)) < 0)
 			perror("sendto failed");
 	}
 
 	json_object_put(result);
 }
 
+void serve_request(struct request_task *task, int sock) {
+	bool compress;
+	struct json_object *result = handle_request(task->request, &compress);
+
+	if (!result)
+		return;
+
+	send_response(
+		sock,
+		result,
+		compress,
+		&task->client_addr
+	);
+}
+
+// Wait for an incomming request and schedule it.
+//
+// 1a. If the schedule is empty, we wait infinite time.
+// 1b. If we have scheduled requests, we only wait for incomming requests
+//     until we reach the scheduling deadline.
+// 2a. If the incomming request was sent to a multicast destination IPv6,
+//     choose a random delay between 0 and max_multicast_delay milliseconds.
+// 2b. If the schedule is full, send the reply immediately.
+// 2c. If the incomming request was sent to a unicast destination, the response
+//     will be also sent immediately.
+static void accept_request(struct request_schedule *schedule, int sock,
+                           uint64_t max_multicast_delay) {
+	char input[REQUEST_MAXLEN];
+	ssize_t input_bytes;
+	struct sockaddr_in6 addr;
+	char control[256];
+	struct in6_addr destaddr;
```

Comment:
Let's initialize destaddr here (`struct in6_addr destaddr = {};`), just to be sure. I seem to remember there might be messages without ip6_pktinfo (using v4-mapped addresses?).